docs/Using Price Feeds/historical-price-data.html

```diff
@@ -29,19 +29,93 @@
+
+
+### `roundId` in Proxy
+
+Because a proxy has references to the current and all previous underlying aggregators , it needs a way to fetch data from the right aggregator. 
+The `roundId` is computed as the following in the [proxy contract](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.7/dev/AggregatorProxy.sol):
+
+```solidity
+roundId = uint80((uint256(phaseId) << 64) | aggregatorRoundId);
+```
+
+where:
+
+- `phaseId` is incremented each time the underlying aggregator implementation has been updated. It is used as key to find the aggregator address. 
+- `aggregatorRoundId` is the aggregator roundId. The id starts at 1.
+- `roundId` computed round id. From the above formula, you can think of it as a large number containing `phaseId` and `aggregatorRoundId`.
+
+> 📘 Remark
+>
+> Note the formula above ensures that no matter how many times the underlying aggregator is changed, the proxy `roundId` will always be increasing.
+>
+
+** Example:**
+
+When querying the historical data, it is important to know when we reach the end of the history of the underlying aggregator. Let's illustrate this with an example:
+At the time of writing, `latestRoundData` [function](/docs/price-feeds-api-reference/#latestrounddata) of  [LINK / USD feed on Ethereum Mainnet](/docs/ethereum-addresses/) returns `roundId = 92233720368547771158`. Using this value, we can compute the `phaseId` and `aggregatorRoundId`:
+
+- `phaseId = 92233720368547771158 >> 64` . Right shifting an integer by 64 bits is equivalent to dividing it by 2^64: `phaseId = 92233720368547771158/ 2^64 = 5` . The current phase id is 5 , which means that this proxy has had 5 underlying aggregators since its initial deployment.
+- `aggregatorRoundId = uint64(92233720368547771158)`. This will retrieve the first 64 bits from the right. If you are calculating this off-chain, you can do the following in javascript:
+
+```javascript
+// First parse to BigInt to perform computation with big integers
+const num = BigInt("92233720368547771158");
+const num2 = BigInt("0xFFFFFFFFFFFFFFFF"); // Largest 64bits integer 
+
+console.log(Number(num >> 64n)); // returns 5 (phaseId)
+console.log(Number(num & num2)) // returns 13078 (aggregatorRoundId) . Use & (AND bitwise operator) which sets each bit to _1_ if both bits are _1_
+```
+
+Now that we have the current aggregator's round _13078_ , we can get its historical data by looping over `getRoundData` [function](/docs/price-feeds-api-reference/#getrounddata) :
+- starting from the first round _92233720368547758081_ (result of _92233720368547771158 - 13078 + 1_).
+- until the current round: _92233720368547771158_. 
+
+_Note_: if we want the historical data for previous aggregators, we have to decrement the `phaseId` and start from round _1_. For phase _4_, we can get the starting roundId off-chain in javascript : 
```

Comment:
```suggestion
To get the historical data for previous aggregators, decrement the `phaseId` and start from round _1_. For phase _4_, get the starting `roundId` off-chain using the following JavaScript example: 
```

"Note" unnecessary unless a callout is required.